src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -429,38 +430,70 @@ protected String addACHAccount(Person payee, PayeeACHAccountExtractDetail achDet
         return processingError;
     }
 
-    /**
-     * Checks if the existing ACH account matches the data fed in from the input file, and creates and routes
-     * a new PAAT document to edit the account if necessary.
-     * 
-     * NOTE: The current implementation just logs match/mismatch status without making updates.
-     * This method will indicate that valid processing occurred whenever it is caled until it
-     * is modified in the future to perform actual updates.
-     */
     protected String updateACHAccountIfNecessary(Person payee, PayeeACHAccountExtractDetail achDetail, PayeeACHAccount achAccount) {
         StringBuilder processingResults = new StringBuilder();
-        processingResults.append("Update functionality NOT implemented yet. Payee \"")
-            .append(achDetail.getNetID()).append("\" data being checked for possible mismatch. ");
-
-        if (!StringUtils.equals(achDetail.getBankRoutingNumber(), achAccount.getBankRoutingNumber())
-                || !StringUtils.equals(achDetail.getBankAccountNumber(), achAccount.getBankAccountNumber())) {
-            /*
-             * For a future enhancement, we will modify this method to actually update the Payee ACH Account
-             * via a maintenance document, and possibly use more fields for change comparison.
-             */
-            processingResults.append(" Account information from input file for payee type '")
-                .append(achAccount.getPayeeIdentifierTypeCode())
-                .append("' does not match what is in KFS, but will be left as-is.");
+
+        if (accountHasChanged(achDetail, achAccount)) {
+            String accountUpdateErrors = updateACHAccount(payee, achDetail, achAccount);
+            processingResults.append(accountUpdateErrors);
         } else {
-            processingResults.append(" Input file's account information for payee of type '")
-                .append(achAccount.getPayeeIdentifierTypeCode())
-                .append("' matches what is already in KFS; no updates will be made for this entry.");
+            LOG.info("Input file's account information for payee of type '" + achAccount.getPayeeIdentifierTypeCode()
+                    + "' matches what is already in KFS; no updates will be made for this entry.");
+        }
+        
+        if (processingResults.length() > 0) {
+            processingResults.append(" Update was NOT performed.");
+            LOG.warn("updateACHAccountIfNecessary: " + processingResults.toString());
         }
-        processingResults.append(" Update was NOT performed.");
-        LOG.warn("updateACHAccountIfNecessary: " + processingResults.toString());
+        
         return processingResults.toString();
     }
 
+    protected boolean accountHasChanged(PayeeACHAccountExtractDetail achDetail, PayeeACHAccount achAccount) {
+        return !StringUtils.equals(achDetail.getBankRoutingNumber(), achAccount.getBankRoutingNumber())

```

Comment:
should this be "EqualsIgnorCase"?

Reply:
Can bank routing numbers be alphanumeric? If they can only be numeric, then I don't think an equalsIgnoreCase check is needed.

Reply:
I think US banks are just numeric, not sure about foreign bank.  It might be worth while checking with Marcia O

Reply:
I checked with Marcia on this. She confirmed that bank routing numbers are always numeric, and that to the best of her knowledge, US bank account numbers are always numeric as well. (She said that foreign accounts are allowed to be alphanumeric, but that only US bank accounts are being used for Cornell's ACH needs, so we don't need to worry about foreign ones in this case.)

Given that, I don't think these routing number and account number checks need an "equalsIgnoreCase" change.

Reply:
works for me, thanks checking